idaes/models_extra/temperature_swing_adsorption/fixed_bed_tsa0d.py

```diff
@@ -0,0 +1,3677 @@
+
+        # scaling factors for compression work
+        for t in self.flowsheet().time:
+            iscale.set_scaling_factor(self.compressor.unit.work_mechanical[t], 1e-5)
+            iscale.set_scaling_factor(self.compressor.unit.work_isentropic[t], 1e-5)
+            iscale.set_scaling_factor(self.compressor.unit.control_volume.work[t], 1e-5)
+
+    def _add_steam_calc(self):
+
+        self.flow_mass_steam = Var(
+            initialize=1.0,
+            domain=PositiveReals,
+            units=units.kg / units.s,
+            doc="Mass flow rate of steam",
+        )
+
+        if self.config.steam_calculation == SteamCalculationType.rigorous:
+
+            # add empty block for heater
+            self.steam_heater = Block()
+
+            # add heater unit model
+            self.steam_heater.unit = Heater(
+                property_package=self.config.steam_properties, has_pressure_change=True
+            )
+
+            # add constraint for total saturation condition
+            @self.steam_heater.unit.Constraint(
+                doc="Constraint outlet stream to be liquid (total saturation)"
+            )
+            def vapor_frac_out_eq(b):
+                return b.control_volume.properties_out[0].vapor_frac == 1e-9
+
+            # add constraints to link heat duty of TSA and heater systems
+            @self.steam_heater.unit.Constraint(doc="Constraint for heat duty of heater")
+            def heat_duty_heater_eq(b):
+                return b.heat_duty[0] == -units.convert(
+                    self.heat_duty_total, to_units=units.J / units.s
+                )
+
+            # fix inlet pressure, enthalpy and pressure drop
+            Pin = 510e3 * units.Pa
+            hin = 52465.8 * units.J / units.mol  # 500 K
+
+            self.steam_heater.unit.inlet.enth_mol[0].fix(hin)
+            self.steam_heater.unit.inlet.pressure[0].fix(Pin)
+            self.steam_heater.unit.deltaP.fix(0)
+
+            # scaling factor for heat duty in heater
+            for t in self.flowsheet().time:
+                iscale.set_scaling_factor(
+                    self.steam_heater.unit.control_volume.heat[t], 1e-5
+                )
+
+        @self.Constraint(doc="Constraint for mass flow rate of steam")
+        def flow_mass_steam_eq(b):
+            if self.config.steam_calculation == SteamCalculationType.simplified:
+                return (
+                    self.flow_mass_steam
+                    == (21.594 * self.heat_duty_total / units.MW + 200.42)
+                    * units.mol
+                    / units.s
+                    * 0.018015
+                    * units.kg
+                    / units.mol
+                )
+
+            if self.config.steam_calculation == SteamCalculationType.rigorous:
+                return (
+                    self.flow_mass_steam
+                    == self.steam_heater.unit.control_volume.properties_out[0].flow_mass
+                )
+
+    def _step_initialize(
```

Comment:
Should this method (and the one that follows) be moved to be with the `Initializer`?

Reply:
Removed along with the old `initialize_build` method